CMakeLists.txt

```diff
@@ -99,6 +101,32 @@ if (BUILD_GUI)
 	)
 endif()
 
+##############################
+# Tests
+##############################
+
+if (BUILD_TESTS)
+	add_executable(
+		tests
+
+		src/tests/image_tests.cpp
+	)
+
+	target_link_libraries(
+		tests PRIVATE
+
+		gtest
```

Comment:
gtest can be downloaded at configure time with FetchContent rather than making it a submodule forcing consumers of the library to download it even if tests are disabled

also use `enable_testing()`

```cmake
include(FetchContent)
FetchContent_Declare(
        googletest
        GIT_REPOSITORY https://github.com/google/googletest.git
        GIT_TAG v1.14.0)
FetchContent_MakeAvailable(googletest)
enable_testing()
```

and if you rename the tests executable to `vtflib_tests` you can put this in

```cmake
target_link_libraries(vtflib_tests PUBLIC gtest_main vtflib)

include(GoogleTest)
gtest_discover_tests(vtflib_tests)
```